torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5665,6 +5821,27 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                SkipInfo('TestJit', 'test_variant_consistency_jit', device_type='cuda',
                         dtypes=(torch.float64, torch.float32), active_if=TEST_WITH_ROCM),
            )),
+    OpInfo('linalg.det',
+           op=torch.linalg.det,
+           variant_test_name='singular',
+           aliases=('det', ),
+           dtypes=double_types(),
+           backward_dtypes=double_types(),
+           aten_name='linalg_det',
+           sample_inputs_func=sample_inputs_linalg_det_singular,
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack, skipCUDAIfRocm],
+           supports_inplace_autograd=False,
+           skips=(
+               SkipInfo('TestCommon', 'test_dtypes'),
+               # the double backward run depends on svd_backward, which is
+               # yet unstable for inputs with repeated singular values
+               SkipInfo('TestGradients', 'test_fn_gradgrad'),
```

Comment:
I agree with eigenvalue solution but only in the context of double backward. I would keep it away from forward as it will cause a significant regression.

Reply:
Absolutely, the suggestion was just for the singular double backward---LU is a much better option in the other setting.

My proposal was just to change `det_backward_singular` from using svd into using `at::linalg_eig`. Something along the lines of (not tested!!):
```c++
Tensor L, V;
std::tie(L, V) = at::linalg_eig(self);

auto L_grad = prod_backward(grad, L, L.prod(-1), -1, false);

return at::linalg_eig_backward({Tensor{}, L_grad, Tensor{}}, self, L, V);
```

I do not think we can check whether double backward can be run in the forward, although cc @albanD just in case.

Reply:
The double backward still fails if replaced with the eig backward. Could it be that the prod_backward is not stable?